{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/components/button.jinja

```diff
@@ -0,0 +1,43 @@
+{% macro button(button_content='Button', dispatch_name='button-click', variant='normal', color='primary', shape='rounded', size='md', type='button') %}
+    {# Variant support: normal, outline #}
+    {# Color support: primary, secondary, accent, success, info, warning, error #}
+    {# Shape support: rounded, square, circle #}
+    {# Size support: xs, sm, md, lg, xl #}
+    {% set class="inline-flex items-center shadow-sm font-medium focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-"+color+"-600" %}
+    {% if size == 'xs' %}
+        {% set class = class + " px-2.5 py-1.5 text-xs" %}
+    {% elif size == 'sm' %}
+        {% set class = class + " px-3 py-2 text-sm" %}
+    {% elif size == 'md' %}
+        {% set class = class + " px-4 py-2 text-sm" %}
+    {% elif size == 'lg' %}
+        {% set class = class + " px-4 py-2 text-base" %}
+    {% elif size == 'xl' %}
+        {% set class = class + " px-6 py-3 text-base" %}
```

Comment:
I can't help but think this would be nicer in a dictionary. I know this code already existed but it's just striking me now.

Reply:
yeah true - lets bug @aidang1 about it :)